janitor/functions.py

```diff
@@ -4327,6 +4327,90 @@ def flag_nulls(
     return df
 
 
+@pf.register_dataframe_method
+def drop_constant_columns(
+    df: pd.DataFrame,
+) -> pd.DataFrame:
+    """
+    Finds and drops the constant columns from a Pandas data frame
+
+    This method does not mutate the original DataFrame.
+    Functional usage syntax:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor as jn
+
+        data_dict = {
+        "a": [1, 1, 1] * 3,
+        "Bell__Chart": [1, 2, 3] * 3,
+        "decorated-elephant": [1, 1, 1] * 3,
+        "animals": ["rabbit", "leopard", "lion"] * 3,
+        "cities": ["Cambridge", "Shanghai", "Basel"] * 3
+        }
+
+        df = pd.DataFrame(data_dict)
+
+        df = jn.functions.drop_constant_columns(df)
+
+    Method chaining usage example:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor
+
+        df = pd.DataFrame(...)
+
+        df = df.drop_constant_columns()
+
+    :param df: Input Pandas dataframe
+    :returns: The Pandas data frame with the constant columns dropped.
+    """
+
+    # :Example 1: Drop columns with a single value:
```

Comment:
What are the commented out lines for?

Reply:
The commented lines below this-> show the example whereas the above show how the function can be called

Reply:
@fireddd @samukweku comments in the code to show examples is a holdover pattern from much before.

Something I learned from _real_ software engineering folks is the following:

1. We have our implementation as a function.
2. Our examples actually live in the test suite, not inside the code.
3. Tutorial-style documentation can reinforce what's in the test suite by bringing them out into the examples gallery.

I can't wait for an in-person sprint at some point to make these a reality and really clean up the codebase. :smile: For now, though, let's leave the comments in-line.

Reply:
I am not sure what's the correct way here. If there is a better way, i would like to do it that way as it would allow me to learn the best practises. For now, i was just trying to replicate whatever was in the existing code.